### DISCUSSION TITLE
Queue status / unperformed queue tasks

### DISCUSSION THREAD

**Discussion by gemathus** [outline/outline#3657] [Self Hosting]
Hi,

Is there a way to look and enqueued tasks in a queue? I'm experiencing a weird behavior where import/export tasks are being created correctly, but are not processing. Whats weird is that some exports perform just fine, but others don't. 

![image](https://user-images.githubusercontent.com/10574406/173419610-36ffd0a7-9a78-4ae8-bf50-358c46385ab3.png)

I'm scaling my Redis instance as we speak, but I don't think that's an elegant fix (if it works) because I'm the only one using the Outline instance now, so I don't think I should need a large Redis instance. 

Any tips as to what might be going on?

P.S. My logs show no errors what so ever.

**Comment by tommoor** [comment]
0 bytes is weird too, it won't be related to the size of redis – it's only used as a message bus. Really need an error to go any further though…

**Reply by gemathus** [reply]
I've been doing some logging, and there are no errors happening. The same is happening on other queues such as de websockets queue. Taking the Websockets queue as an example:

Everything works well until `await websocketQueue.add(job.data)` in the worker service. But for some reason, it stops there, it doesn't go to `websocketQueue.process()` in the websockets service. It get's added to the queue, but it never processes.

**Reply by tommoor** [reply]
Sounds like there is no worker process running at all to be honest.

Workers pick jobs off the queue to process, months ago imports did not happen on the worker so is probably why that worked before.

**Reply by gemathus** [reply]
No, it's running because a lot of jobs do get processed correctly. It's only every now and then that it fails (without an error. It just looses de job).
But it just occurred to me. I'm running all services, including collaboration, in the same box, which defaults concurrency to 1. Do you think this might be causing an issue?

**Reply by tommoor** [reply]
> I'm running all services, including collaboration, in the same box, which defaults concurrency to 1. Do you think this might be causing an issue?

Highly unlikely, most folks run it like this – multiple boxes are only needed for scale in the thousands of users. Please post full logs from the moment of import until ~5minutes or so afterwards

**Reply by gemathus** [reply]
I added a few additional logs trying to debug the problem. As you'll see, there's an entry with the message `scheduled ImportMarkdownZipTask with id e751d7c0-3a12-429b-9f3b-87cd446b2e6e`, which is the last mention of the task. After that you'll see if does not get processed. 

[log-events-viewer-result.csv](https://github.com/outline/outline/files/8910170/log-events-viewer-result.csv)

**Reply by gemathus** [reply]
BTW, you might notice that its running on localhost:3000, but that's by design since its running on AWS ECS containers behind a load balancer with HTTPS listeners.

**Reply by tommoor** [reply]
Yea, still unclear but as your codebase is forked the possibilities are endless. One other debugging thing, try `NODE_ENV=bull` which should output what's happening on the queue
